dist/images/Dockerfile.dpdk1911

```diff
@@ -0,0 +1,72 @@
+# This package is required to build the OVS-DPDK RPMs and is available in Fedora 32.
+FROM fedora:32
+
+ENV RPM_ARCH=x86_64
+ENV ARCH=amd64
+
+ENV OVN_VERSION=20.03
+ENV OVS_VERSION=2.13.0
+
+ENV DPDK_VERSION=19.11.1
+ENV DPDK_TARGET=x86_64-native-linuxapp-gcc
+
+ENV OVN_DIR=/usr/src/ovn
+ENV OVS_DIR=/usr/src/openvswitch-${OVS_VERSION}
+ENV DPDK_DIR=/usr/src/dpdk-stable-${DPDK_VERSION}
+
+ENV PATH=${PATH}:/usr/share/openvswitch/scripts
+ENV PATH=${PATH}:/usr/share/ovn/scripts/
+
+RUN dnf install -y wget xz git make numactl-devel diffutils dnf-plugins-core dpdk-devel libcap-ng-devel libpcap-devel nc iptables ipset hostname && dnf clean all
+
+# NOTE: Fedora 32 comes with gcc version 10. This caused LD linker issues during DPDK build.
+# Install gcc version 9 and its dependencies.
+RUN mkdir /rpms && cd /rpms
+RUN wget https://kojipkgs.fedoraproject.org//packages/gcc/9.2.1/1.fc32.3/x86_64/cpp-9.2.1-1.fc32.3.x86_64.rpm \
+         https://kojipkgs.fedoraproject.org//packages/gcc/9.2.1/1.fc32.3/x86_64/libgomp-9.2.1-1.fc32.3.x86_64.rpm \
+         https://kojipkgs.fedoraproject.org//packages/gcc/9.2.1/1.fc32.3/x86_64/gcc-9.2.1-1.fc32.3.x86_64.rpm
+RUN dnf install -y cpp-9.2.1-1.fc32.3.x86_64.rpm libgomp-9.2.1-1.fc32.3.x86_64.rpm gcc-9.2.1-1.fc32.3.x86_64.rpm
+RUN rm -f cpp-9.2.1-1.fc32.3.x86_64.rpm libgomp-9.2.1-1.fc32.3.x86_64.rpm gcc-9.2.1-1.fc32.3.x86_64.rpm
+
+# Install DPDK
+RUN cd /usr/src/ && \
+  wget http://fast.dpdk.org/rel/dpdk-${DPDK_VERSION}.tar.xz && \
+  tar xf dpdk-${DPDK_VERSION}.tar.xz && \
+  rm -f dpdk-${DPDK_VERSION}.tar.xz && \
+  cd ${DPDK_DIR} && \
+  sed -i s/CONFIG_RTE_EAL_IGB_UIO=y/CONFIG_RTE_EAL_IGB_UIO=n/ config/common_linux && \
+  sed -i s/CONFIG_RTE_LIBRTE_KNI=y/CONFIG_RTE_LIBRTE_KNI=n/ config/common_linux && \
+  sed -i s/CONFIG_RTE_KNI_KMOD=y/CONFIG_RTE_KNI_KMOD=n/ config/common_linux && \
+  make install T=${DPDK_TARGET} DESTDIR=install
+
+# Install OVS-DPDK
+# NOTE: rpm-build is installed here, as installing it earlier conflicts with the workaround to roll back the gcc version.
+RUN dnf install -y rpm-build
+RUN cd /usr/src/ && \
+  wget https://www.openvswitch.org/releases/openvswitch-${OVS_VERSION}.tar.gz && \
+  tar xf openvswitch-${OVS_VERSION}.tar.gz && \
+  rm -f openvswitch-${OVS_VERSION}.tar.gz && \
+  cd ${OVS_DIR} && \
+  sed -e 's/@VERSION@/0.0.1/' rhel/openvswitch-fedora.spec.in > /tmp/ovs.spec && \
+  dnf builddep -y /tmp/ovs.spec && \
+  ./boot.sh && \
+  ./configure --prefix=/usr/ --with-dpdk=${DPDK_DIR}/${DPDK_TARGET} && \
+  make rpm-fedora RPMBUILD_OPT="--with dpdk --without check"
+
+# Install OVN
+RUN cd /usr/src/ && \
+  git clone -b branch-${OVN_VERSION} --depth=1 https://github.com/ovn-org/ovn.git && \
+  cd ovn && \
+  ./boot.sh && \
+  ./configure --with-ovs-source=${OVS_DIR} && \
+  make rpm-fedora
+
+RUN cp ${OVS_DIR}/rpm/rpmbuild/RPMS/${RPM_ARCH}/* ${OVN_DIR}/rpm/rpmbuild/RPMS/${RPM_ARCH}/* /rpms
+
+RUN mkdir -p /var/run/openvswitch && \
+    mkdir -p /var/run/ovn
+
```

Comment:
I find the final image is 3.12GB can we use multi-stage build that mv all the rpm packages in a new Fedora image to install and reduce the image size?

Reply:
I’ve tried this. To install OVS-DPDK, you must first install DPDK in the environment, then build and install OVS against this DPDK install. Unfortunately, the resulting RPMs don’t seem to be portable, even if another environment already has same DPDK install.
I guess this is the reason you must build and install OVS-DPDK locally instead of it being available as RPM/YUM/DNF.